src/bus/webSocketConnection.js

```diff
@@ -9,10 +9,89 @@ class WebSocketConnection {
    * @param {string} organizationId UUID corresponding with an organization
    */
   constructor(webSocket, organizationId) {
+    this._jsonRpcId = 0;
     this._organizationId = organizationId;
     this._webSocket = webSocket;
   }
 
+  /**
+   * Sends a message to the message bus to authorize a channel
+   *
+   * @param {string} token JSON Web Signature containing the channel and actions needed for authorization
+   *
+   * @returns {Promise}
+   * @fulfill {Object} Whether you are authorized or not and any error
+   * @reject {errorEvent} The error event from the WebSocket
+   *
+   * @example
+   * axios.post(['https://', SERVICE_ADDRESS, '/authorizations/channels'].join(''), {
+   *   resources: [
+   *     {
+   *       resource: 'CHANNEL_YOU_NEED',
+   *       actions: ['publish', 'subscribe']
+   *     }
+   *   ]
+   * }, {
+   *   headers: {
+   *     Authorization: ['Bearer', ACCESS_TOKEN_FROM_PREVIOUS_STEP].join(' ')
+   *   }
+   * }).then((response) => {
+   *   const token = response.data;
+   *
+   *   contxtSdk.bus.connect('4f0e51c6-728b-4892-9863-6d002e61204d')
+   *     .then((webSocket) => {
+   *       webSocket.authorize(token).then((res) => {
+   *         if(res.error) {
+   *          console.log("Error: ", res.error)
+   *         }
+   *
+   *         console.log(res.authorized)
+   *       })
+   *       .catch((authErrorEvent) => {
+   *         console.log(authErrorEvent)
+   *       });
+   *     })
+   *     .catch((errorEvent) => {
+   *       console.log(errorEvent);
+   *     });
+   * });
+   */
+  authorize(token) {
+    return new Promise((resolve, reject) => {
+      if (!token) {
+        return reject(new Error('A token is required for authorization'));
+      }
+
+      if (!this._webSocket || !this._webSocket.OPEN) {
+        return reject(new Error('WebSocket connection not open'));
+      }
+
+      this._webSocket.onmessage = (message) => {
+        const messageData = JSON.parse(message.data);
+        const error = messageData.error || null;
+        const authorized = !error;
+
+        return resolve({ authorized, error });
```

Comment:
Can you make this `isAuthorized`?

Reply:
I think you might need to tear down these onmessage/onerror handlers too

Reply:
If there's an error, would it be more appropriate to reject?

Reply:
I went back and forth with the error resolve/reject for a while. It was kinda weird because if you're unauthorized it comes back in the "error" even though there isn't technically an error

Reply:
Since this method abstracts away the message bus portion of things, it's weird to me that it wouldn't reject. You wouldn't expect your promise's `then` to get an error

Reply:
Yeah that makes sense. Will update